### DISCUSSION TITLE
[Next.js] Is there any way to use getStaticProps?

### DISCUSSION THREAD

**Discussion by sevenc-nanashi** [markdoc/markdoc#149] [Q&A] [answered]
Hello!
I'm making my blog with markdoc, but is there any way to use getStaticProps?
```tsx
export async function getStaticProps(_context) {
  return {
    props: {
      test: "a",
    },
  }
}

export default function App({ Component, pageProps }: AppProps) {
  const { markdoc, test } = pageProps

  console.log(test)  // => undefined
  // ...
}
```
I want to use it for static generation.

**Comment by mfix-stripe** [comment, marked as answer]
@sevenc-nanashi can you share more about what you are trying to accomplish? You can't _explicitly_ use `getStaticProps` within a Markdoc file, but each file generates its own `getStaticProps`. 

For example, you can pass values via the Markdoc frontmatter, and access them in `_app.js`  via `pageProps.markdoc.frontmatter`

**Reply by sevenc-nanashi** [reply]
I want to run command in getStaticProps of each article.

https://github.com/sevenc-nanashi/blog/blob/68e9ca0660d84e374c0ff00858345442a52f72e1/pages/index.tsx#L29
This is what I'm making, this gets the creation time from git.

**Reply by mfix-stripe** [reply]
@sevenc-nanashi you can create a custom `document` [Node](https://markdoc.dev/docs/nodes) and call your command in the transform function. 

Here is an example of a custom `document` node: https://github.com/markdoc/docs/blob/main/markdoc/nodes/document.markdoc.js
